### PR TITLE
Cleaning up unnecessary code from the CI yaml and CI server version upgrade

### PR DESCRIPTION
**Pull request by ohbus**
- [x] Added **`-y`** flag for installing **`xvfb`**

- [x] Removed the switching using `if` that I previously forgot to eliminate from the script which was useful for using the same file for push and pull requests from **Java CI**

- [x] Removed the **`if`** checking in the Pull Requests builder as well.

- [x] upgraded CI server version from Ubuntu 18.04 to **20.04**

### CODE REVIEW

File: .github/workflows/maven-ci.yml [hunk 1]
```diff
@@ -49,14 +49,10 @@ jobs:
           ${{ runner.os }}-maven-
     # Some tests need screen access
     - name: Install xvfb
-      run: sudo apt-get install xvfb
+      run: sudo apt-get install -y xvfb
     # SonarQube scan does not work for forked repositories
     # See https://jira.sonarsource.com/browse/MMF-1371
```
Comment (iluwatar) [inline .github/workflows/maven-ci.yml:54]: I would extend the comment here to say that we run the SonarQube analysis only for the `master` branch.

Comment (iluwatar) [reply .github/workflows/maven-ci.yml:54]: And point out that there is a separate workflow for pull requests.

File: .github/workflows/maven-pr-builder.yml [hunk 2]
```diff
@@ -49,9 +49,8 @@ jobs:
           ${{ runner.os }}-maven-
     # Some tests need screen access
     - name: Install xvfb
-      run: sudo apt-get install xvfb
+      run: sudo apt-get install -y xvfb
     # SonarQube scan does not work for forked repositories
     # See https://jira.sonarsource.com/browse/MMF-1371
```
Comment (iluwatar) [inline .github/workflows/maven-pr-builder.yml:54]: Similarly we should here elaborate that this workflow only runs for incoming pull requests and there is a separate workflow for `master` branch.

**Review by iluwatar — CHANGES_REQUESTED** [review]
Thanks for the improvements! I also suggest couple of additions to comments so we can make this really shine.

File: .github/workflows/maven-ci.yml (hunk 1 above)
Comment (ohbus) [reply .github/workflows/maven-ci.yml:54]: Good suggestion!

File: .github/workflows/maven-pr-builder.yml (hunk 2 above)
Comment (ohbus) [reply .github/workflows/maven-pr-builder.yml:54]: Great!

**Comment by ohbus** [comment]
@iluwatar Shall we also upgrade the server version from Ubuntu 18.04 to **20.04**?

**Comment by iluwatar** [comment]
@ohbus why not, go ahead

**Comment by ohbus** [comment]
Okay @iluwatar  Let me commit the changes and let's try that out.

**Comment by ohbus** [comment]
The PR builder went smoothly! Check [here](https://github.com/iluwatar/java-design-patterns/runs/1045287215).

Let's add the the commit to master and see if it creates any problem.
If it does these will be the commit hash, to help us revert them.

- [x] [f588632](https://github.com/iluwatar/java-design-patterns/pull/1509/commits/f5886325ec8e07e994f4eeda8d7a4adb1e391f45)

- [x] [1973d1b](https://github.com/iluwatar/java-design-patterns/pull/1509/commits/1973d1bc63bc60727b5fdba0dda9070eb66f5cd9)

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Thanks @ohbus, so far looks good 👍🏻

**Pull request merged**